fish-rust/src/job_group.rs

```diff
@@ -146,11 +147,9 @@ impl JobGroup {
     }
 
     /// Gets the cancellation signal, if any.
-    pub fn get_cancel_signal(&self) -> Option<NonZeroI32> {
-        match self.signal.load(Ordering::Relaxed) {
-            0 => None,
-            s => Some(NonZeroI32::new(s).unwrap()),
-        }
+    pub fn get_cancel_signal(&self) -> Option<Signal> {
+        let s = self.signal.load(Ordering::Relaxed);
+        Signal::try_new(s as i32)
```

Comment:
This makes it look like the Signal conversion may "fail." We should always return a signal if `self.signal` is nonzero.